java/ql/src/experimental/Security/CWE/CWE-927/SensitiveBroadcast.qhelp

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>Broadcast intents in an Android application are visible to all applications installed on the same mobile device, exposing all sensitive information they contain.</p>
+<p>Broadcasts are vulnerable to passive eavesdropping or active denial of service attacks when an intent is broadcast without specifying any receiver permission or receiver application.</p>
+</overview>
+
+<recommendation>
+<p>Specify a receiver permission or application when broadcasting intents, or switch to <code>LocalBroadcastManager</code> or the latest <code>LiveData</code> library.</p>
+</recommendation>
+
+<example>
+<p>The following example shows two ways of broadcasting intents. In the 'BAD' case, no "receiver permission" is specified. In the 'GOOD' case, "receiver permission" or "receiver application" is specified.</p>
+<sample src="SensitiveBroadcast.java" />
+</example>
+
+<references>
+<li>
+<a href="https://cwe.mitre.org/data/definitions/927.html">CWE-927: Use of Implicit Intent for Sensitive Communication</a>
```

Comment:
Could we add some references here? I'm thinking perhaps something general about Intents, and then perhaps some reference to the mentioned `LocalBroadcastManager` and `LiveData`.

Reply:
Thanks @aschackmull for reviewing this PR. I've added four more references, which cover intents, broadcasts using implicit intent, security implications, best practices like using <code>LocalBroadcastManager</code>, and the <code>LiveData</code> architecture.